Sources/MotorControl/ViewModel/MotorControlAssessmentViewModel.swift

```diff
@@ -0,0 +1,132 @@
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import SwiftUI
+import AssessmentModelUI
+import AssessmentModel
+import SharedResources
+
+public final class MotorControlAssessmentViewModel : AssessmentViewModel {
+//    override public func nodeState(for node: Node) -> NodeState? {
+//        if let step = node as? HandInstructionObject,
+//           let whichHand = HandSelection(rawValue: currentBranchState.node.identifier) {
+//            return TwoHandInstructionState(step, whichHand: whichHand, parentId: currentBranchState.id)
+//        }
+//        else {
+//            return super.nodeState(for: node)
+//        }
+//    }
+    
+    override public func nodeState(for node: Node) -> NodeState? {
+        let nodeState = super.nodeState(for: node)
+        if let instructionState = nodeState as? InstructionState {
+            if let imageInfo = instructionState.contentNode.imageInfo as? AnimatedImage {
+                iterateAnimatedImage(instructionState, imageInfo)
+            }
+            if let whichHand = HandSelection(rawValue: currentBranchState.node.identifier) {
+                swapPlaceholderStringsAndReverseImage(in: instructionState, with: whichHand)
+            }
+        }
+//        else if let overviewNode = node as? OverviewStepObject {
+//            let overviewState = OverviewState(overviewNode)
+//            overviewState.subtitle = "In overview state"
+//            if let imageInfo = overviewState.contentNode.imageInfo as? AnimatedImage {
+//                iterateAnimatedImage(overviewState, imageInfo)
+//            }
+//            return overviewState
+//        }
+        return nodeState
+    }
+    
+    private func swapPlaceholderStringsAndReverseImage(in instructionState: InstructionState, with whichHand: HandSelection) {
+        let handPlaceHolder = "%@"
+        instructionState.title = instructionState.title?.replacingOccurrences(of: handPlaceHolder, with: whichHand.rawValue.uppercased())
+        instructionState.subtitle = instructionState.subtitle?.replacingOccurrences(of: handPlaceHolder, with: whichHand.rawValue.uppercased())
+        instructionState.detail = instructionState.detail?.replacingOccurrences(of: handPlaceHolder, with: whichHand.rawValue.uppercased())
+        
+        if whichHand.rawValue == HandSelection.right.rawValue, let imageInfo = instructionState.contentNode.imageInfo as? FetchableImage, let uiImage = UIImage(named: imageInfo.imageName, in: SharedResources.bundle, compatibleWith: nil) {
+            instructionState.image = Image(uiImage: uiImage.withHorizontallyFlippedOrientation())
+        }
```

Comment:
This won't work for the images - need to move this to the InstructionView since you're using the ContentImage there or mutate the image info or use your own NodeState implementations that include which hand, etc.

Reply:
Implemented logic for mirroring an image in the InstructionView.